### PR TITLE
Considered timezone

### PR DESCRIPTION
**Pull request by konifar**
https://github.com/konifar/droidkaigi2016/pull/273/files#r52994581

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java [hunk 1]
```diff
@@ -103,6 +103,20 @@ public static Date getDisplayDate(@NonNull Date date, Context context) {
         }
     }
 
+    public static Date getConfTimezoneCurrentDate() {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: I confirmed by debug log that same date print in the case of different timezone.

```
UTC: Tue Feb 16 20:43:08 UTC 2016
Japan Standard Time: Tue Feb 16 20:44:12 GMT+09:00 2016
```

Comment (hotchemi) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: OK, should we create DateFormat instance per each method call? Or make them constant.

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: Sounds good. I'll refactor as you said! Thanks!

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: Like this?

```java
    private static DateFormat dateFormatTokyo;
    private static DateFormat dateFormatLocal;
    
    public static Date getConfTimezoneCurrentDate() {
        Date date = new Date();
        try {
            return getLocalDateFormat().parse(getTokyoDateFormat().format(date));
        } catch (ParseException e) {
            Log.e(TAG, "date: " + date + "can not parse." + e);
            return date;
        }
    }

    private static DateFormat getTokyoDateFormat() {
        if (dateFormatTokyo == null) {
            dateFormatTokyo = SimpleDateFormat.getDateTimeInstance();
            dateFormatTokyo.setTimeZone(CONFERENCE_TIMEZONE);
        }
        return dateFormatTokyo;
    }

    private static DateFormat getLocalDateFormat() {
        if (dateFormatLocal == null) {
            dateFormatLocal = SimpleDateFormat.getDateTimeInstance();
            dateFormatLocal.setTimeZone(TimeZone.getDefault());
        }
        return dateFormatLocal;
    }
```

Comment (hotchemi) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: Ah, thx. But I guess you can initialize them at the declaration time.

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: I think so too. But `setTimezone()` have to be called every time.

```
    private static final DateFormat DATE_FORMAT_TOKYO = SimpleDateFormat.getDateTimeInstance();
    private static final DateFormat DATE_FORMAT_LOCAL = SimpleDateFormat.getDateTimeInstance();

    public static Date getConfTimezoneCurrentDate() {
        DATE_FORMAT_TOKYO.setTimeZone(CONFERENCE_TIMEZONE);
        DATE_FORMAT_LOCAL.setTimeZone(TimeZone.getDefault());
        Date date = new Date();
        try {
            return DATE_FORMAT_LOCAL.parse(DATE_FORMAT_TOKYO.format(date));
        } catch (ParseException e) {
            Log.e(TAG, "date: " + date + "can not parse." + e);
            return date;
        }
    }
```

Comment (hotchemi) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: ‘‘‘java 
private static final DateFormat dateFormatTokyo =  SimpleDateFormat.getDateTimeInstance();
‘‘‘

Comment (hotchemi) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: Oh really? I'm not sure but if it is, you don't have to fix, Thx.

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: OK! Thanks!

Comment (hotchemi) [inline app/src/main/java/io/github/droidkaigi/confsched/util/LocaleUtil.java]: Sorry, now I'm home.

I wanted to say like below. setTimeZone would be called at the first time LocaleUtil being loaded.
And you can cache DateFormat instances on the memory:D

```java
    private static final DateFormat DATE_FORMAT_TOKYO = SimpleDateFormat.getDateTimeInstance();
    private static final DateFormat DATE_FORMAT_LOCAL = SimpleDateFormat.getDateTimeInstance();
    static {
           DATE_FORMAT_TOKYO.setTimeZone(CONFERENCE_TIMEZONE);
           DATE_FORMAT_LOCAL.setTimeZone(TimeZone.getDefault());
    }
    public static Date getConfTimezoneCurrentDate() {
        Date date = new Date();
        try {
            return DATE_FORMAT_LOCAL.parse(DATE_FORMAT_TOKYO.format(date));
        } catch (ParseException e) {
            Log.e(TAG, "date: " + date + "can not parse." + e);
            return date;
        }
    }
```

**Pull request merged**